vividus-extension-selenium/src/main/java/org/vividus/monitor/GenericHarOnFailureMonitor.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2019-2021 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.vividus.monitor;
+
+import java.util.Optional;
+
+import javax.inject.Inject;
+
+import com.browserup.harreader.model.Har;
+import com.google.common.eventbus.EventBus;
+
+import org.vividus.proxy.IProxy;
+import org.vividus.util.json.JsonUtils;
+
+public class GenericHarOnFailureMonitor extends AbstractHarOnFailureMonitor
+{
+    @Inject private IProxy proxy;
+
+    public GenericHarOnFailureMonitor(EventBus eventBus, JsonUtils jsonUtils)
+    {
+        super(eventBus, jsonUtils);
+    }
+
+    @Override
+    protected Optional<Har> takeAssertionFailureHar()
+    {
+        Har har = proxy.getRecordedData();
+        proxy.clearRecordedData();
```

Comment:
Doubtful solution, we may break tests

Reply:
Do you mean to clear the recorded data every time after an unsuccessful step in general? Or the method itself?

Reply:
users may check the recorded data to validate requests/responses, if the data is cleared on failure, the tests may produce false-negative results